guides-markdown/connect-local-nodes.md

```diff
@@ -0,0 +1,153 @@
+# Connect two local nodes
+
+```post-author
+Orfeas Stefanos Thyfronitis Litos
+```
+
+```post-description
+Start and manually connect two regtest nodes in a single script.
+```
+
+When we start two nodes in the regtest (loopback network, 127.0.0.1) with
+manually set ports, they don't get connected automatically because they don't
+know each other's port. We thus have to connect them manually. In this example
+we will fire up one SPV and one Full node and initiate a connection from the
+SPV side. The SPV node is given the address of the Full node and attempts to
+connect. If everything goes well, the two nodes connect. Finally they are
+gracefully closed.
+
+The script should work out of the box, the only requirement is having `bcoin`
+installed.
+
+```javascript
+const bcoin = require('bcoin').set('regtest');
+const NetAddress = bcoin.net.NetAddress;
+const Network = bcoin.Network;
+const pEvent = require('p-event'); // tool to await for events
+
+async function delay(ms) {
+  return new Promise(resolve => {
+    setTimeout(resolve, ms);
+  });
+}
+
+const regtest = Network.get().toString();
+
+// create nodes
+const spvNode = new bcoin.SPVNode({
+  network: regtest,
+  httpPort: 48449, // avoid clash of ports
+
+  // write log file and chain data to specific directory
+  prefix: '~/connect-test/SPV',
+  memory: false,
+  logFile: true,
+  logConsole: false,
+  logLevel: 'spam',
+
+  // reduce log spam on SPV node (won't warn on Full node)
```

Comment:
sorry this is my typo, it should say `won't work on full node` -- the full node will still try to connect to that 1 outbound peer and the log will fill up anyway. `maxOutbound: 0` returns an error, I tried it :-)

Reply:
Ah, now I see what you meant. Maybe we can achieve the intended behaviour by making the full node `selfish`. I'll try it and add it if it reduces logging but still broadcasts stuff (I suppose it should).

Reply:
I will add a comment explaining why `selfish` and why users shouldn't do it in general :)

Reply:
I thought `selfish` prevented all blocks and txs from going out...? My intent with the `maxOutbound` was just to reduce the `attempt to connect to peer` stuff in the log... it's not that crucial either way :-)

Reply:
Apparently `selfish` mode doesn't work. First of all, selfish nodes do not connect to spv nodes. Second, even if I try to undo some of the damage done by the `selfish` flag, apparently the handshake fails with the (selfish) full node never sending back the `verack` packet... Probably what I tried to do is unsupported, but still it's weird behavior. Anyway, let's forget the `selfish` trick.